data/persistence/src/androidMain/kotlin/social/androiddev/common/persistence/di/persistenceModule.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.common.persistence.di
+
+import com.russhwolf.settings.SharedPreferencesSettings
+import com.squareup.sqldelight.android.AndroidSqliteDriver
+import kotlinx.serialization.json.Json
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import social.androiddev.common.persistence.AuthenticationDatabase
+import social.androiddev.common.persistence.localstorage.DodoAuthStorage
+import social.androiddev.common.persistence.localstorage.DodoAuthStorageImpl
+
+/**
+ * Koin DI module for all android specific persistence dependencies
+ */
+actual val persistenceModule: Module = module {
+    single<DodoAuthStorage> {
```

Comment:
I notice the paramterized type is delcared here but not in the single on line 39. Is that just style or is there any significance?

Reply:
Yes it's intentional, the type below would be redundant and is not required. The koin bean definition can infer the type fine.

However I am specifying the `interface` as the type here, otherwise the inferred type would be the implementation class and any DI that looks for resolving the `interface` would fail.